### PR TITLE
Fix resend email issue AB#7995 and AB#8098

### PR DESCRIPTION
**Pull request by sslaws**
## Status
Ready

## Fixes or Implements [AB#7995](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/7995), [AB#8098](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/8098)
- [ ] Enhancement
- [X] Bug
- [ ] Documentation

## Description:
An email was not being sent when a user tried to resend the email validation.  This was due to a comparison between the email being passed in and the profile email not matching (as the profile email was null since it wasn't validated).  I simplified the logic and also add a expiry check to the validation logic.

## Testing
- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [X] Not Required

### Steps to Reproduce:
After registering with an email if you hit the resend email in profile it would not send.

### UI Changes
Yes

### Browsers Tested
- [X] Chrome
- [ ] Safari
- [ ] Edge
- [ ] Firefox

Provide an explanation for any test(s) not completed.
Test will be performed in subsequent task.

## Notes/Additional Comments
Any addtional notes or comments that may be relevant.  Include any specialized deployment steps here.

### CODE REVIEW

File: Apps/WebClient/src/ClientApp/app/views/timeline.vue [hunk 1]
```diff
@@ -1,657 +1,638 @@
 <style lang="scss" scoped>
-@import "@/assets/scss/_variables.scss";
```
Comment (sslaws) [inline Apps/WebClient/src/ClientApp/app/views/timeline.vue:2]: I only made changes on line 96 for new text in banner.  The changes were likely caused by lintfix which I did run as per normal routine.

File: Apps/WebClient/src/ClientApp/app/views/timeline.vue [hunk 2]
```diff
@@ -388,20 +391,26 @@ export default class TimelineComponent extends Vue {
   }
 
   private get getNumberOfEntriesPerPage(): number {
-    if (this.windowWidth < 576) { // xs
```
Comment (FuriousLlama) [inline Apps/WebClient/src/ClientApp/app/views/timeline.vue:391]: these numbers need to be constants. I'am guessing this is not part of the issue fix. Lets create a ticket to address this.

File: Apps/WebClient/src/ClientApp/app/views/timeline.vue [hunk 3]
```diff
@@ -388,20 +391,26 @@ export default class TimelineComponent extends Vue {
...
     } // else, xl
     return 15;
   }
 
   private get numberOfPages(): number {
-    let result = Math.ceil(this.filteredEntriesLength / this.getNumberOfEntriesPerPage);
+    let result = Math.ceil(
+      this.filteredEntriesLength / this.getNumberOfEntriesPerPage
+    );
     if (result < 1) {
```
Comment (FuriousLlama) [inline Apps/WebClient/src/ClientApp/app/views/timeline.vue:414]: this should be impossible/necessary

File: Apps/WebClient/src/Server/Services/UserEmailService.cs [hunk 4]
```diff
@@ -83,56 +82,27 @@ public EmailInvite RetrieveLastInvite(string hdid)
...
         {
             this.logger.LogTrace($"Updating user email...");
-            bool retVal = false;
             UserProfile userProfile = this.profileDelegate.GetUserProfile(hdid).Payload;
-            EmailInvite emailInvite = this.emailInviteDelegate.GetLastForUser(hdid);
+            EmailInvite emailInvite = this.RetrieveLastInvite(hdid);
 
-            if (email != userProfile.Email)
+            this.logger.LogInformation($"Removing email from user ${hdid}");
+            userProfile.Email = null;
```
Comment (FuriousLlama) [inline Apps/WebClient/src/Server/Services/UserEmailService.cs:89]: looks much cleaner! Glad you where able to make it more succinct

File: Apps/WebClient/src/ClientApp/app/views/timeline.vue (hunk 2 above)
Comment (sslaws) [reply Apps/WebClient/src/ClientApp/app/views/timeline.vue:391]: Agreed AB#8168 created.

**Pull request merged**